deps/wazuh_testing/wazuh_testing/scripts/statistical_data_analyzer.py

```diff
@@ -0,0 +1,339 @@
+    dataframe = load_dataframe(datasource)
+    alpha = float((100 - percentage) / 100)
+    daemons = dataframe_ref['Daemon'].unique()
+
+    for daemon in daemons:
+        print(f"\nDaemon: {daemon}\n")
+        ref_daemon_data = dataframe_ref[dataframe_ref['Daemon'] == daemon]
+        new_daemon_data = dataframe[dataframe['Daemon'] == daemon]
+
+        for value in STATS_MAPPING.values():
+            ref_values = ref_daemon_data[value].dropna()
+            new_values = new_daemon_data[value].dropna()
+
+            # t-Student
+            t_stat, t_p_value = ttest_ind(ref_values, new_values, equal_var=False)
+            # Levene
+            l_stat, l_p_value = levene(ref_values, new_values)
+            # Anova
+            a_stat, a_p_value = f_oneway(ref_values, new_values)
+
+            print(f"\nMetric '{value}':")
+            print(f"    - t-Student Test: t-Statistic={t_stat}, p-value={t_p_value}")
+            print(f"    - Levene's Test: Statistic={l_stat}, p-value={l_p_value}")
+            print(f"    - ANOVA's Test: Statistic={a_stat}, p-value={a_p_value}")
+
+            if t_p_value < alpha:
+                print(f"\n  Significant difference detected in '{value}' with {percentage}% confidence (t-Test).")
+            if l_p_value < alpha:
+                print(f"\n  Variance difference detected in '{value}' with {percentage}% confidence (Levene's Test).")
+            if a_p_value < alpha:
+                print(f"\n  Means difference detected in '{value}' with {percentage}% confidence (ANOVA's Test).")
+        print("\n")
+
+
+def print_dataframes_stats(baseline, datasource):
+    """Print a PrettyTable with the statistics for each daemon and metric.
+
+    Args:
+        baseline: Dataframe with the baseline values.
+        datasource: Dataframe with the values to compare.
+    """
+    daemons = baseline['Daemon'].unique()
+    output = ""
+
+    for daemon in daemons:
+        baseline_data = baseline[baseline['Daemon'] == daemon]
+        datasource_data = datasource[datasource['Daemon'] == daemon]
+
+        for value in STATS_MAPPING.values():
+            table = PrettyTable()
+            table.title = daemon + " - " + value
+            table.field_names = ['Name', 'Mean', 'Max value', 'Min value', 'Standard deviation', 'Variance']
+            table.add_row(["Baseline", round(baseline_data[value].mean(), 2),
+                        baseline_data[value].max(), baseline_data[value].min(),
+                        round(baseline_data[value].std(), 2), round(baseline_data[value].var(), 2)])
+            table.add_row(["Data source", round(datasource_data[value].mean(), 2),
+                        datasource_data[value].max(), datasource_data[value].min(),
+                        round(datasource_data[value].std(), 2), round(datasource_data[value].var(), 2)])
+            output += table.get_string() + "\n\n"
+    
+    return output
+
+
+def calculate_percentage_change(dataframe_ref, dataframe):  
```

Comment:
It's better to change the current approach to allow more modular analysis, allowing review metric by metric instead of a monolithic approach.

Reply:
This was an old function whose use does not make sense in the new design. It has been removed in https://github.com/wazuh/wazuh-qa/pull/5622/commits/26dc5cfa8d7f2339d13d5d5820d1ad7cceda3625